pkg/engine/datasource/graphql_datasource/graphql_datasource_test.go

```diff
@@ -8146,6 +8146,26 @@ func TestSource_Load(t *testing.T) {
 			assert.Equal(t, `{"variables":{"a":null,"b":"b","c":{}}}`, buf.String())
 		})
 	})
+	t.Run("remove undefined variables", func(t *testing.T) {
+		var (
+			src       = &Source{httpClient: &http.Client{}}
+			serverUrl = ts.URL
+			variables = []byte(`{"a":null,"b":"null"}`)
```

Comment:
`"b":"null"` is a string `"null"` not the null value
I believe it should be `"b": null`

In case you will set it like this and set a flag `UNNULLVARIABLES` both `a` and `b` will be removed

Reply:
typo